### PR TITLE
`merge` -> `deep-merge`

### LINKED ISSUE #52: Fields are not merged
This query:
```graphql
query { 
  viewer { 
    login
  }
  viewer {
    databaseId
  }
}
```

Should return something like:
```edn
{:data {:viewer {:login "someone" :id "qwertyuiop"}}}
```

Instead, it returns only the `databaseId`. If we query for the login last, it returns just the login.

### PR DESCRIPTION
**Pull request by bcarrell**
Minor change to ensure fields merge when they should.  Refer to the Issue for more details.

Closes #52

### CODE REVIEW

File: test/com/walmartlabs/lacinia_test.clj [hunk 1]
```diff
@@ -236,6 +254,16 @@
              }
            }"]
     (is (= {:data {:luke {:name "Luke Skywalker"}}}
+           (execute default-schema q nil nil))))
+  (let [q "query FetchLukeAliased {
+             luke: human(id: \"1000\") {
```
Comment (hlship) [inline test/com/walmartlabs/lacinia_test.clj]: If I read the spec correctly, this is valid as long as all the arguments to `human` are the same. Do we check that? I think we need a validation and test to ensure that operation uniqueness is maintained.

File: src/com/walmartlabs/lacinia/executor.clj [hunk 2]
```diff
@@ -267,9 +267,9 @@
     (apply-selection execution-context selection)))
 
 (defn ^:private combine-map-results
-  "Left associative resolution of results, combined using merge."
+  "Left associative resolution of results, combined using deep-merge."
```
Comment (hlship) [inline src/com/walmartlabs/lacinia/executor.clj]: There's some cases I'd be worried about that may not be handled by deep-merge, such as:

```
{ human { friends { name } }
  human { friends { id } } }
```

Which should produce `{:data {:human {:friends [{:name ... :id ...} ...]}}}`.  I'd like to see a test for that.

**Comment by hlship** [comment]
I'm wondering if you explored merging together the field selections, rather than deep merging the data?  The former is complicated BUT would be more efficient, I suspect.  I'm going to pull this branch and run the benchmarks.

File: src/com/walmartlabs/lacinia/executor.clj (hunk 2 above)
Comment (bcarrell) [inline src/com/walmartlabs/lacinia/executor.clj]: Yeah this doesn't work.  Fixing

**Comment by bcarrell** [comment]
> I'm wondering if you explored merging together the field selections, rather than deep merging the data? The former is complicated BUT would be more efficient, I suspect. I'm going to pull this branch and run the benchmarks.

Looked briefly at doing it like this; it seemed like a complex way of going about it vs. merging the data.  Unsure the perf differences will be worth the effort and complexity.

File: test/com/walmartlabs/lacinia_test.clj (hunk 1 above)
Comment (bcarrell) [inline test/com/walmartlabs/lacinia_test.clj]: The case where the arguments are different isn't handled, at least as part of this PR.  This case should be a validation error (pre-execution).

**Comment by hlship** [comment]
Ran the benchmarks, no appreciable difference in parse or execution time.

**Comment by bcarrell** [comment]
Gonna close this for a bit while working on some stuff.

**Comment by gusbicalho** [comment]
This solution will run the resolver multiple times, right?

**Comment by bcarrell** [comment]
I'm going to explore doing this as part of parse instead of after execution has taken place to avoid the resolver issue.

**Pull request closed without merging**